codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/testutil/Rust.kt

```diff
@@ -55,8 +56,18 @@ private fun tempDir(directory: File? = null): File {
  * This workspace significantly improves test performance by sharing dependencies between different tests.
  */
 object TestWorkspace {
-    private val baseDir =
-        System.getenv("SMITHY_TEST_WORKSPACE")?.let { File(it) } ?: tempDir()
+    private val baseDir by lazy {
+        val homeDir = System.getProperty("APPDATA")
+            ?: System.getenv("XDG_DATA_HOME")
+            ?: System.getProperty("user.home")
+                ?.let { Path.of(it, "Library/Application Support").absolutePathString() }
```

Comment:
Shouldn't this be `Path.of(it, ".local/share").absolutePathString()`?

Reply:
Yep, `Library/Application Support` is for Mac I presume.

Reply:
AFAIK programs following the XDG spec use `~/.local/share` on Mac too (at least on my machine, things like `neovim`, `nix`, and `ruby` do).